cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -760,11 +758,15 @@ public static void updateStatus(Throwable error, KafkaConnector kafkaConnector2,
         Set<Condition> unknownAndDeprecatedConditions = validate(connector);
         unknownAndDeprecatedConditions.forEach(condition -> conditions.add(condition));
 
-        StatusUtils.setStatusConditionAndObservedGeneration(connector, status, error != null ? Future.failedFuture(error) : Future.succeededFuture());
-        status.setConnectorStatus(statusResult);
+        if (connector.getMetadata().getGeneration() != null)    {
+            status.setObservedGeneration(connector.getMetadata().getGeneration());
```

Comment:
Same as the other comment. Shouldn't this be the observed generation from the previous status?